5/23/19. Re-started pipeline from raw reads for Caballo Moro, Jineo, Escondido, and new samples from Josh Gross and UMGC ("June2019" samples).

Used Adamâ€™s pipeline and Adam's alignments for all other individuals (https://github.umn.edu/aherman/cavefish_RIS).

################################################################################################

Step 1: FastQC, adapter removal and quality trimming 


All raw reads were quality checked with FastQC and processed with cutadapt-1.2.1 and trimmomatic-0.30.

Samples from BGI (Jineo, Escondido, and all other samples from JG) had unknown barcodes (kept private by BGI), so they were processed with cutadapt using the following Illumina TruSeq adapters and NNNs in place of the individual barcode on R1 (need to supply wildcard flag to cutadapt for this to work):
R1: GATCGGAAGAGCACACGTCTGAACTCCAGTCACNNNNNNATCTCGTATGCCGTCTTCTGCTTG
R2: AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT

See the following scripts for examples of how reads were processed: 
par_fastqc.sh
par_Gross_samples_cutadapt.sh
par_Gross_samples_trimmomatic.sh

Caballo Moro samples had the same adapters as BGI samples, but barcodes on R1 were known for each individual and supplied to cutadapt. Cutadapt version 1.2.1 and the newest version (2.3.0) both had issues processing these samples (kept resulting in corrupt fastqs with extra lines), so trimmomatic-0.30 was used to do the adapter clipping and quality trimming for these samples (see par_CabMoro_trimmomatic.sh).

After processing with cutadapt and/or trimmomatic, unpaired reads from R1 and R2 files need to be combined into one fastq file (see par_combine_unpair.sh).

Reads are then ready to align.


################################################################################################

Step 2: Alignment to surface and cave genomes


For each sample, processed reads were aligned to both the cavefish (v1) and the surface fish (v2) genomes.

Used array jobs rather than parallel for alignments to be more efficient. 

The script make_align_jobs_all_raw.sh will generate jobs (a list of commands, one for each sample) that can be specified in the array script array_map.sh. Run the following command to get the list of commands writted to the file raw_bwa_commands_cavefish.txt (can modify make_align_jobs_all_raw.sh for alignment to cave or sruface genome):

./make_align_jobs_all_raw.sh > raw_bwa_commands_cavefish.txt

Then need to edit array_map.sh to point to raw_bwa_commands_cavefish.txt. To submit the array job, use the following command:

qsub -t 1-N array_map.sh 

where N is equal to the number of jobs in the array (i.e. the number of lines in the commands .txt file generated by make_align_jobs_all_raw.sh).


Use picard to mark duplicates and add read group (i.e. sample name) for each raw bam. 


Use samtools view to split bams into mapped and unmapped reads.


Used mapped bams for gatk haplotype caller.



